setup.py

```diff
@@ -104,16 +104,17 @@ def build_extension(self, ext):
             env.get('CXXFLAGS', ''),
             self.distribution.get_version()
         )
-        if not os.path.exists(self.build_temp):
-            os.makedirs(self.build_temp)
+        build_dir = os.path.join(self.build_temp, dims)
+        if not os.path.exists(build_dir):
```

Comment:
This can be simplified so that the explicit 'if' is not needed:

os.makedirs(build_dir, exist_ok=True)